src/main/java/com/frcteam3636/frc2024/RemoteADStarAK.java

```diff
@@ -0,0 +1,150 @@
+        }
+
+        Logger.processInputs("RemoteADStarAK", io);
+
+        if (io.currentPathPoints.isEmpty()) {
+            return null;
+        }
+
+        return PathPlannerPath.fromPathPoints(io.currentPathPoints, constraints, goalEndState);
+    }
+
+    /**
+     * Set the start position to pathfind from
+     *
+     * @param startPosition Start position on the field. If this is within an obstacle it will be
+     *                      moved to the nearest non-obstacle node.
+     */
+    @Override
+    public void setStartPosition(Translation2d startPosition) {
+        if (!Logger.hasReplaySource()) {
+            io.adStar.setStartPosition(startPosition);
+        }
+    }
+
+    /**
+     * Set the goal position to pathfind to
+     *
+     * @param goalPosition Goal position on the field. f this is within an obstacle it will be moved
+     *                     to the nearest non-obstacle node.
+     */
+    @Override
+    public void setGoalPosition(Translation2d goalPosition) {
+        if (!Logger.hasReplaySource()) {
+            io.adStar.setGoalPosition(goalPosition);
+        }
+    }
+
+    /**
+     * Set the dynamic obstacles that should be avoided while pathfinding.
+     *
+     * @param obs             A List of Translation2d pairs representing obstacles. Each Translation2d represents
+     *                        opposite corners of a bounding box.
+     * @param currentRobotPos The current position of the robot. This is needed to change the start
+     *                        position of the path to properly avoid obstacles
+     */
+    @Override
+    public void setDynamicObstacles(List<Pair<Translation2d, Translation2d>> obs, Translation2d currentRobotPos) {
+        if (!Logger.hasReplaySource()) {
+            io.adStar.setDynamicObstacles(obs, currentRobotPos);
+        }
+    }
+
+    private static class ADStarIO implements LoggableInputs {
+        public RemoteADStar adStar = new RemoteADStar();
+        public boolean isNewPathAvailable = false;
+        public List<PathPoint> currentPathPoints = Collections.emptyList();
+
+        @Override
+        public void toLog(LogTable table) {
+            table.put("IsNewPathAvailable", isNewPathAvailable);
+
+            double[] pointsLogged = new double[currentPathPoints.size() * 2];
+            int idx = 0;
+            for (PathPoint point : currentPathPoints) {
+                pointsLogged[idx] = point.position.getX();
+                pointsLogged[idx + 1] = point.position.getY();
+                idx += 2;
+            }
+
+            table.put("CurrentPathPoints", pointsLogged);
+        }
+
+        @Override
+        public void fromLog(LogTable table) {
+            isNewPathAvailable = table.get("IsNewPathAvailable", false);
+
+            double[] pointsLogged = table.get("CurrentPathPoints", new double[0]);
+
+            List<PathPoint> pathPoints = new ArrayList<>();
+            for (int i = 0; i < pointsLogged.length; i += 2) {
```

Comment:
2? What is 2?

Reply:
cause when the points are added it adds x point and y point, so to go to the next logged point i += 2